src/hocon.erl

```diff
@@ -434,6 +466,21 @@ nested_put([Key], Val, Map) ->
 nested_put([Key|Paths], Val, Map) ->
     merge(Key, nested_put(Paths, Val, #{}), Map).
 
+nested_update_with(Fun, Map) ->
+    nested_update_with(Fun, Map, maps:iterator(Map)).
+
+nested_update_with(Fun, Map, I) ->
```

Comment:
we may want to support converting Key from `atom()` to `binary()` or `string()`,
so it's better to create another Map instead of updating in-place. ?

Reply:
fixed